clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/BaseOperation.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store.operations;
+
+/**
+ * Base class that represents a {@link org.ehcache.Cache} operation
+ *
+ * @param <K> key type
+ */
+public abstract class BaseOperation<K> {
+
+  public enum Code {
```

Comment:
What I think Louis has in mind is to augment the `OperationCodecFactory` class to contain a pair of maps: one `Map<Class<? extends BaseOperation>,? extends OperationCodec>` for encoding where you can lookup the codec using the operation and another one `Map<Byte, ? extends OperationCodec>` for decoding where you can lookup the codec using the op byte.

That way, you'd get rid of the need to use switch blocks as that would centralize that logic in a single location.